tools/logparser2.py

```diff
@@ -111,6 +111,36 @@ def _meta(log):
             yield sensor, os.path.basename(filepath)
 
 
+def _merge(streams):
+    vector = []
+    valid = []
+    for s in streams:
+        try:
+            item = s.next()
+            vector.append(item)
+            valid.append(s)
+        except StopIteration:
+            pass
+
+    while valid:
+        i = _min_index(vector)
+        yield vector[i]
+        try:
+            vector[i] = valid[i].next()
+        except StopIteration:
+            del vector[i]
+            del valid[i]
+
+
+def _min_index(vector):
+    index, value = 0, vector[0]
```

Comment:
expects non-zero vector ... maybe for these two functions could be nice unit test demonstrating your idea that identical timestamps do not change order, for example??

Reply:
```
>>> min([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: min() arg is an empty sequence
```
or
```
>>> import numpy as np
>>> np.argmin([1,0])
1
>>> np.argmin([])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\Python27\lib\site-packages\numpy\core\fromnumeric.py", line 948, in a
rgmin
    return _wrapit(a, 'argmin', axis)
  File "c:\Python27\lib\site-packages\numpy\core\fromnumeric.py", line 43, in _w
rapit
    result = getattr(asarray(obj), method)(*args, **kwds)
ValueError: attempt to get argmin of an empty sequence
```
... I am not sure if it is worth it (as private method), but test would be nice anyway ...

Reply:
It is used only internally and it is never called with empty list. When the vector is empty, the mergesort is done. I don't see the need to accept empty list.